### ISSUE TITLE
Date comparison

### ISSUE THREAD

**Issue by geoo89** [africanmathsinitiative/R-Instat#7303]
**Describe the bug**
I want to filter rows having a date entry before/after a certain date. It's not clear whether this works or is possible.

**To Reproduce**
Steps to reproduce the behaviour:
1. Create a column with date-times in ISO format (it appears that it is detected as a date)
![image](https://user-images.githubusercontent.com/678199/158342953-563c339d-d0e9-4f9a-b7c2-d62cfc609bef.png)
2. Create a filter to only pick rows where the date entry is after a certain date
![image](https://user-images.githubusercontent.com/678199/158342792-6d3115f9-859e-47d5-93eb-44393e4a1a5b.png)
![image](https://user-images.githubusercontent.com/678199/158342816-46633835-70bc-45e1-ae8e-fa398cff695a.png)
3. After applying the filter, only the NAs are filtered out, not dates before the cutoff.
![image](https://user-images.githubusercontent.com/678199/158342495-1b755671-c89a-4dc4-b637-aa2bdf9019b7.png)

**Expected behaviour**
Only show rows where the date is after the cutoff.

**Screenshots**
If applicable, add screenshots to help explain your problem, particularly of error messages.

**R script or commands**
![image](https://user-images.githubusercontent.com/678199/158342535-35f0a05e-8d82-43c9-8e38-4dfe24360b41.png)

**Details (optional):**
 - OS: Ubuntu 18.04 (AppImage version: https://downloads.r-instat.org/R-Instat-latest-x86_64.AppImage)

**Comment by rdstern** [comment]
@geoo89 you are right, we don't yet cope well with data-time variables.  As a work-around you could use Prepare > Column: Date > Make Date, and put in your date-time variable, to produce (say) date1, that is just a date variable.  Then the filter should work better on that variable.

**Comment by geoo89** [comment]
Example file:

[date_example.csv](https://github.com/africanmathsinitiative/R-Instat/files/8253447/date_example.csv)

**Comment by Patowhiz** [comment]
@rdstern another important question is why or how R-Instat recognised the column as a date column on import, notice the D in the column name. But in the filter dialog it's recognised as a numeric, notice the presence of the numeric number pad.
@geoo89 thanks for supplying the sample data.

**Comment by Patowhiz** [comment]
This issue relates to Issue #6685.

**Comment by rdstern** [comment]
@Patowhiz as I men tioned before, that's because it is a date-time vartiable.  I defined the date from that and it ius recognised

![image](https://user-images.githubusercontent.com/11226469/158391417-582935d3-a2a6-4700-b0b2-6f61c523505a.png)
